WcaOnRails/app/models/competition.rb

```diff
@@ -1561,6 +1572,15 @@ def self.search(query, params: {}, managed_by_user: nil)
       competitions = Competition.visible
     end
 
+    if params[:continent].present?
+      continent = Continent.find(params[:continent])
+      if !continent
```

Comment:
`if !x` can always be replaced with `unless x`.
That being said, you can also make it single line like:
`raise WcaExceptions::BadApiParameter.new("Invalid continent: '#{params[:continent]}'") unless continent.nil?`.

Also, if you do `Continent.find!` instead of `Continent.find`, then it will automatically raise exception and you don't have to do it explicitly.

Reply:
The existing code for other parameters follow a similar syntax of checking `if !x` and then `raise`, should I just simplify all of them or keep them as is and follow the convention myself?

Reply:
I don't see a clean way of doing it this way if we want to keep the exceptions for each parameter..